gammapy/stats/feldman_cousins.py

```diff
@@ -0,0 +1,442 @@
+        largest_entry = np.where(distributions_re_scaled < 1, distributions_re_scaled, -1)
+        # For each mu, this is the position of the largest entry that is not yet a rank.
+        largest_entry_position = np.argmax(largest_entry, axis = 1)
+        # Invalidate indices where there is no maximum (every entry is already a rank)
+        largest_entry_position = [largest_entry_position[i] if largest_entry[i][largest_entry_position[i]] != -1 \
+                                                            else -1 for i in range(len(largest_entry_position))]
+        # Replace the largest entry with the highest rank so far plus one
+        # Add the probability
+        for i in range(number_mus):
+            if largest_entry_position[i] == -1:
+                continue
+            distributions_re_scaled[i][largest_entry_position[i]] = largest_rank[i] + 1
+            if summed_propability[i] < alpha:
+                summed_propability[i] += distributions_scaled[i][largest_entry_position[i]]
+                distributions_scaled[i][largest_entry_position[i]] = 1
+            else:
+                distributions_scaled[i][largest_entry_position[i]] = 0
+
+    return distributions_scaled
+
+
+def fc_get_upper_and_lower_limit(mu_bins, x_bins, acceptance_intervals):
+    """Find upper and lower limit from acceptance_intervals.
+
+    Parameters
+    ----------
+    mu_bins : array-like
+        The bins used in mue direction.
+    x_bins : array-like
+        The bins of the x distribution
+    acceptance_intervals : array-like
+        The output of fc_construct_acceptance_intervals_pdfs.
+
+    Returns
+    -------
+    upper_limit : array-like
+        Feldman Cousins upper limit x-coordinates
+    lower_limit : array-like
+        Feldman Cousins lower limit x-coordinates
+    x_values: array-like
+        All the points that are inside the acceptance intervals
+    """
+
+    upper_limit = []
+    lower_limit = []
+    x_values    = []
+
+    number_mu     = len(mu_bins)
+    number_bins_x = len(x_bins)
+
+    for mu in range(number_mu):
+        upper_limit.append(-1)
+        lower_limit.append(-1)
+        x_values.append([])
+        acceptance_interval = acceptance_intervals[mu]
+        for x in range(number_bins_x):
+            #This point lies in the acceptance interval
+            if acceptance_interval[x] == 1:
+                x_value = x_bins[x]
+                x_values[-1].append(x_value)
+                # Upper limit is the first point where this condition is true
+                if upper_limit[-1] == -1:
+                    upper_limit[-1] = x_value
+                # Lower limit is the first point after this condition is not true
+                if x == number_bins_x - 1:
+                    lower_limit[-1] = x_value
+                else:
+                    lower_limit[-1] = x_bins[x + 1]
+
+    return upper_limit, lower_limit, x_values
+
+
+def fc_fix_upper_and_lower_limit(upper_limit, lower_limit):
```

Comment:
The order (upper, lower) is a bit unintuitive, no?
Is there a reason you prefer it over (lower, upper)?

If no, my suggestion would be we use the (lower, upper) order everywhere (not just for fc) in Gammapy,
and then IMO it would also be OK to shorten this function name to `fc_fix_limits`.

If you want to stick with the (upper, lower) order, I think it's good to stick with the long function name that makes the order explicit.